tools/execute-unit-tests-with-coverage

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+# Warning: kcov is a Linux only tool
+
+set -ex
+
+KCOV_VERSION="30"
+
+CURRENT_PATH="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+PROJECT_HOME="$CURRENT_PATH/.."
+KCOV_TEMP="$PROJECT_HOME/target/kcov"
+KCOV_HOME="$KCOV_TEMP/kcov-$KCOV_VERSION"
+KCOV_BINARY="$KCOV_HOME/src/kcov"
+
+
+get_and_compile_kcov_locally() {
+  mkdir -p "$KCOV_TEMP"
```

Comment:
Since we always build the same kcov version, maybe we should host a pre-built version to cut testing times? It's fine to do that in a followup.